src/Impostor.Api.Innersloth.Generator/Vector2Converter.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Numerics;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Impostor.Api.Innersloth.Generator;
+
+public class Vector2Converter : JsonConverter<Vector2>
+{
+    public override Vector2 Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        if (reader.TokenType != JsonTokenType.StartObject) throw new JsonException();
+
+        float x = 0;
+        float y = 0;
```

Comment:
is it a valid vector2 if x or y are omitted from the json object?

Reply:
I'd say it doesn't really matter. And for the default json converter it would also default to 0 if missing unless `[JsonRequired]` is specified.